cns/restserver/ipam.go

```diff
@@ -76,6 +97,66 @@ func (service *HTTPRestService) requestIPConfigHandler(w http.ResponseWriter, r
 	logger.ResponseEx(service.Name+operationName, ipconfigRequest, reserveResp, reserveResp.Response.ReturnCode, err)
 }
 
+var errStoreEmpty = errors.New("empty endpoint state store")
+
+func (service *HTTPRestService) updateEndpointState(ipconfigRequest cns.IPConfigRequest, podInfo cns.PodInfo, podIPInfo cns.PodIpInfo) error {
+	if service.EndpointStateStore == nil {
+		return errStoreEmpty
+	}
+	service.Lock()
+	defer service.Unlock()
+	logger.Printf("[updateEndpointState] Updating endpoint state for infra container %s", ipconfigRequest.InfraContainerID)
+	if endpointInfo, ok := service.EndpointState[ipconfigRequest.InfraContainerID]; ok {
+		logger.Printf("[updateEndpointState] Found existing endpoint state for infra container %s", ipconfigRequest.InfraContainerID)
+		ip, ipnet, err := net.ParseCIDR(podIPInfo.PodIPConfig.IPAddress + "/" + fmt.Sprint(podIPInfo.PodIPConfig.PrefixLength))
+		if err != nil {
+			return fmt.Errorf("failed to parse pod ip address: %w", err)
+		}
+		ipconfig := net.IPNet{IP: ip, Mask: ipnet.Mask}
```

Comment:
instead of this, there s better way to do it.
```
ip := net.ParseIP(podIPInfo.PodIPConfig.IPAddress)
if ip.To4() == nil {
   ipconfig := net.IPNet{IP: ip, Mask: net.CIDRMask(podIPInfo.PodIPConfig.PrefixLength, 32)}
} else {
  ipconfig := net.IPNet{IP: ip, Mask: net.CIDRMask(podIPInfo.PodIPConfig.PrefixLength, 128)}
}
```
}

Reply:
this looks cleaner! Will make the change.